aten/src/ATen/native/TensorShape.cpp

```diff
@@ -834,6 +836,25 @@ Tensor make_qtensor(const Tensor& self, IntArrayRef size, IntArrayRef stride, Qu
 }
 
 Tensor as_strided_tensorimpl(const Tensor& self, IntArrayRef size, IntArrayRef stride, optional<int64_t> storage_offset_) {
+
+
+  TensorImpl* tensor_impl = self.unsafeGetTensorImpl();
+  torch::lazy::LTCTensorImpl* ltc_tensor_impl = dynamic_cast<torch::lazy::LTCTensorImpl*>(tensor_impl);
+  //printf("as_strided_tensorimpl is tensor impl: %p? is LTC tensor impl: %p?\n", tensor_impl, ltc_tensor_impl);
+  if (ltc_tensor_impl) {
+    auto self2 = ltc_tensor_impl->tensor()->ToTensor(false);
```

Comment:
self2 should be a legitimate eager-cuda tensor in every respect.  I'm not sure why you are doing stuff like making a storage, view based on it; could you perhaps just return self2?

note: ToTensor(false) means lazy tensor isn't makign you a separate copy, and if you (eager you) does something evil, it could essentially corrupt the storage that lazy is counting on for doing future computations.  This may be OK but you have to be sure that you're only calling it this way in a time when you are going permanently back to eager and won't use the lazy tensor again.

Reply:
> note: ToTensor(false) means lazy tensor isn't makign you a separate copy, and if you (eager you) does something evil, it could essentially corrupt the storage that lazy is counting on for doing future computations. This may be OK but you have to be sure that you're only calling it this way in a time when you are going permanently back to eager and won't use the lazy tensor again.

Yes, this part is well understood.

> self2 should be a legitimate eager-cuda tensor in every respect. I'm not sure why you are doing stuff like making a storage, view based on it; could you perhaps just return self2?

I'm just simply copying what `as_strided_tensorimpl` does for a regular tensor.
The problem I'm getting here is that, `self2` apparently is NOT a legitimate eager-cuda tensor and it does NOT contain storage. Maybe in my hack I accidentally messed something else up?!
Note: in my initial prototype based on a much older lazy_tensor_staging commit, this actually was the case (self2 does have storage). I don't know what changed in recent code base but I only notice the breakage after deleting the other branch and now I don't have a working example any more.

:sob:

Reply:
oh jeez. Well, i think i can say with confidence that self2 should have storage, and hopefully this is some kind of rebase/integration bug.